client/src/partials/templates/modals/findReference.modal.js

```diff
@@ -102,24 +102,18 @@ function FindReferenceModalController(Instance, Debtor, Creditor, Voucher, Filte
           filter : { condition : filtering.byDate }
         },
       ];
-      /**
-       * @fixme: the voucher API returns two lines
-       * for credit and debit with the same amount; and we need
-       * only one line with the amount
-       * we choose to discare the credit line
-       */
+
+      // format data for the grid
       var data = [];
       list.forEach(function (item) {
```

Comment:
Since you no longer have to do a `if` statement, this is much better expressed as a `.map()`.

```js
var data = list.map(function (item) {
  return {
   uuid : item.uuid,
   /* other props ... */
  };
});

vm.gridOptions.data = data;
```

This comment will not block this PR.